src/librustc_mir/dataflow/impls/borrows.rs

```diff
@@ -38,7 +38,7 @@ pub struct Borrows<'a, 'gcx: 'tcx, 'tcx: 'a> {
     location_map: FxHashMap<Location, BorrowIndex>,
     region_map: FxHashMap<Region<'tcx>, FxHashSet<BorrowIndex>>,
     region_span_map: FxHashMap<RegionKind, Span>,
-    nonlexical_regioncx: Option<&'a RegionInferenceContext>,
+    nonlexical_regioncx: Option<RegionInferenceContext>,
```

Comment:
@nikomatsakis I'm not sure if this will be an issue in future work for NLL. Should I add the comments from gitter here?

Reply:
The problem, I believe, is that `FlowInProgress<T>` is invariant with respect to `T`. That means that it doesn't allow any lifetimes in `T` to be approximated. This is unfortunate, but ok, not much we can do easily do about it. This is also why you have to introduce `'a` above, rather than just relying on `'cx`. 

In any case, because of that, when we have a `FlowInProgress<Borrows<'a...>>` and a `FlowInProgress<MoveUninits<'a...>>` in the same function call (as you do now), then the lifetime `'a` for both of them has to be precisely the same. This means it has to live as long as the longest thing, which then lives longer than the variable `nonlexical_regioncx`.

Anyway, taking ownership is actually fine.